### PR TITLE
Fix cid parsing

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

Fix CID parsing if string is less then 16 characters.

## Type of change

- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [x] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m [hunk 1]
```diff
@@ -51,33 +45,19 @@ - (NSData *)msalStringAsGUIDData
     
     if (stringLen > 16)
     {
+        MSID_LOG_WITH_CTX(MSIDLogLevelWarning, nil, @"Failed to parse unsigned long long from a string, length: %d", stringLen);
```
Comment (Veena11) [inline MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m:48]: can it be less that 16 ?

File: MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m [hunk 2]
```diff
@@ -51,33 +45,19 @@ - (NSData *)msalStringAsGUIDData
...
     
-    NSUInteger zeroFillLen = (16 - (stringLen + 1) / 2);
-    NSMutableData *result = [[NSMutableData alloc] initWithLength:16];
-    
-    char chars[3] = {'\0','\0','\0'};
-    
-    for (int i = stringLen-1; i >= 0; i-=2)
+    NSScanner *scanner = [NSScanner scannerWithString:self];
+    unsigned long long ull;
+    if (![scanner scanHexLongLong:&ull])
```
Comment (Veena11) [inline MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m:54]: For my learning, why do we use scanHexLongLong and not scanHexInt. Is it an unsigned long long value we are reading from the string ?

File: MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m (hunk 1 above)
Comment (antrix1989) [reply MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m:48]: yep

File: MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m (hunk 2 above)
Comment (antrix1989) [reply MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m:54]: The value won't fit into Int, ests sends the long long value to us.

Comment (Veena11) [reply MSAL/src/configuration/external/ios/NSString+MSALAccountIdenfiers.m:54]: Thank you Serhii

**Review by Veena11 — APPROVED** [review]

**Pull request merged**